Sources/Confidence/EventStorage.swift

```diff
@@ -0,0 +1,127 @@
+    private var currentFileHandle: FileHandle?
+
+    init() throws {
+        self.folderURL = try EventStorageImpl.getFolderURL()
+        if !FileManager.default.fileExists(atPath: folderURL.backport.path) {
+            try FileManager.default.createDirectory(at: folderURL, withIntermediateDirectories: true)
+        }
+        try resetCurrentFile()
+    }
+
+    func startNewBatch() throws {
+        try storageQueue.sync {
+            guard let currentFileName = self.currentFileUrl else {
+                return
+            }
+            try currentFileHandle?.close()
+            try FileManager.default.moveItem(at: currentFileName, to: currentFileName.appendingPathExtension(READYTOSENDEXTENSION))
+            try resetCurrentFile()
+        }
+    }
+
+    func writeEvent(event: Event) throws {
+        try storageQueue.sync {
+            guard let currentFileHandle = currentFileHandle else {
+                return
+            }
+            let encoder = JSONEncoder()
+            let serialied = try encoder.encode(event)
+            let delimiter = "\n".data(using: .utf8)
+            guard let delimiter else {
+                return
+            }
+            currentFileHandle.seekToEndOfFile()
+            try currentFileHandle.write(contentsOf: delimiter)
+            try currentFileHandle.write(contentsOf: serialied)
+        }
+    }
+
+
+    func batchReadyIds() throws -> [String] {
+        try storageQueue.sync {
+            let fileUrls = try FileManager.default.contentsOfDirectory(at: folderURL, includingPropertiesForKeys: nil)
+            return fileUrls.filter({ url in url.pathExtension == READYTOSENDEXTENSION }).map({ url in url.lastPathComponent })
+        }
+    }
+
+    func eventsFrom(id: String) throws -> [Event] {
+        try storageQueue.sync {
+            let decoder = JSONDecoder()
+            let fileUrl = folderURL.appendingPathComponent(id)
+            let data = try Data(contentsOf: fileUrl)
+            let dataString = String(data: data, encoding: .utf8)
+            return try dataString?.components(separatedBy: "\n")
+                .filter({ events in !events.isEmpty })
+                .map({ eventString in try decoder.decode(Event.self, from: eventString.data(using: .utf8)!) }) ?? []
+        }
+    }
+
+    func remove(id: String) throws {
+        try storageQueue.sync {
+                let fileUrl = folderURL.appendingPathComponent(id)
+                try FileManager.default.removeItem(at: fileUrl)
+        }
+    }
+
+    private func getLastWritingFile() throws -> URL? {
+        let files = try FileManager.default.contentsOfDirectory(at: folderURL, includingPropertiesForKeys: nil)
+        for fileUrl in files {
+            if fileUrl.pathExtension != READYTOSENDEXTENSION {
+                return fileUrl
+            }
+        }
+        return nil
+    }
+
+    private func resetCurrentFile() throws {
+        if let currentFile = try getLastWritingFile() {
```

Comment:
Leave a comment here about this handling the case of a currently unclose file from a different session?